### DISCUSSION TITLE
Is complex JQL supported?

### DISCUSSION THREAD

**Discussion by pmatulis** [ankitpokhrel/jira-cli#663] [Q&A] [answered]
I tried copying and pasting a complex JQL query from a working filter in the Jira GUI into the `-q` option and it couldn't cope with it. Is complex JQL supported? The JQL was of the form:

```
project in (BLAH) AND component = "FOO BAR" AND labels = LABEL AND sprint IN openSprints() ORDER BY status ASC,statusCategory
```

Output included:

```
Error:
  - The value 'FOO' does not exist for the field 'component'.
```

I got most of it to work. This does:

```
project in (BLAH) AND component = 'FOO BAR' AND labels = LABEL AND sprint IN openSprints() 
```

Notice the single quotes. So I can't get `ORDER BY status ASC,statusCategory` to work:

```
Error:
  - Error in the JQL Query: Expecting either 'ASC' or 'DESC' but got 'ORDER'. (line 1, character 144)
```

**Comment by danpritts** [comment, marked as answer]
Seems likely that this is a shell quoting error.  For *nix, best practice is probably to avoid `'` in your query, and then wrap the entire query in `'`.

Maybe the feature request should be to be able to pull the JQL from a file.

**Comment by ankitpokhrel** [comment]
This is indeed a shell quoting error. Check this doc for more info: https://www.etalabs.net/sh_tricks.html (especially "Shell-quoting arbitrary strings" section).
